spellcheck.pl

```diff
@@ -135,30 +140,39 @@ sub spellcheck_key_pressed
 
     return unless Irssi::settings_get_bool('spellcheck_enabled');
 
-    # I know no way to *mark* misspelled words in the input line,
-    # that's why there's no spellcheck_print_suggestions -
-    # because printing suggestions is our only choice.
-
     # hide correction window when message is sent
     if (chr($key) =~ /\A[\r\n]\z/ && $correction_window) {
         $correction_window->command("window hide $window_name");
+        if (Irssi->can('gui_input_clear_extents')) {
+            Irssi::gui_input_clear_extents(0, 9999);
+        }
+    }
+
+    # get current inputline
+    my $inputline = Irssi::parse_special('$L');
+    if (Irssi::settings_get_str('term_charset') =~ /utf-?8/i) {
+        Encode::_utf8_on($inputline);
+    }
+    if (Irssi->can('gui_input_set_extent')) {
+	Irssi::gui_input_set_extent(length $inputline, '%n');
     }
 
     # don't bother unless pressed key is space or dot
     return unless (chr $key eq ' ' or chr $key eq '.');
 
-    # get current inputline
-    my $inputline = Irssi::parse_special('$L');
+    $inputline = substr $inputline, 0, Irssi::gui_input_get_pos;
 
     # check if inputline starts with any of cmdchars
     # we shouldn't spell-check commands
-    my $cmdchars = Irssi::settings_get_str('cmdchars');
-    my $re = qr/^[$cmdchars]/;
-    return if ($inputline =~ $re);
+    return unless -1 == index(Irssi::settings_get_str('cmdchars'), substr($inputline, 0, 1));
 
     # get last bit from the inputline
-    my ($word) = $inputline =~ /\s*(\S+)$/;
+    my ($word) = $inputline =~ /\s*(\S+)[.\s]*$/;
```

Comment:
Why?

Reply:
the input line will now end with what we typed, so we should ignore/allow space and .

Reply:
Ignoring the dot doesn't work, because it will be consumed by `\S+`. But meh, we can fix it later.